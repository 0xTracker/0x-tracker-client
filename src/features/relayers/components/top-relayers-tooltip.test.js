import { render } from 'react-testing-library';
import React from 'react';

import TopRelayersTooltip from './top-relayers-tooltip';

it('should render null without payload', () => {
  const { container } = render(<TopRelayersTooltip />);

  expect(container.firstChild).toBeNull();
});

it('should render null with empty payload', () => {
  const { container } = render(<TopRelayersTooltip payload={[]} />);

  expect(container.firstChild).toBeNull();
});

it('should render with payload', () => {
  const { container } = render(
    <TopRelayersTooltip
      currency="USD"
      payload={[
        {
          payload: {
            relayer: { name: 'Radar Relay' },
            share: 22.5,
            trades: 50,
            volume: 540600,
          },
        },
      ]}
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('should render for ETH currency', () => {
  const { container } = render(
    <TopRelayersTooltip
      currency="ETH"
      payload={[
        {
          payload: {
            relayer: { name: 'Radar Relay' },
            share: 50,
            trades: 50,
            volume: 540600,
          },
        },
      ]}
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});
