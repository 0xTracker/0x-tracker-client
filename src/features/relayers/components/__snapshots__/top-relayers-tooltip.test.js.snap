// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with payload 1`] = `
.c0 {
  background-color: #f7f8fe;
  border-radius: 0.25rem;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  padding: 1rem;
  text-align: left;
}

.c1 {
  font-weight: bold;
  font-size: 1rem;
  margin-bottom: 1.2rem;
  text-align: center;
}

.c3 {
  display: inline-block;
  font-weight: normal;
}

.c4 {
  display: inline-block;
  margin-left: 0.25rem;
}

.c2 {
  margin: 0;
  padding: 0;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Radar Relay
  </h1>
  <dl
    class="c2"
  >
    <div>
      <dt
        class="c3"
      >
        market share
        :
      </dt>
      <dd
        class="c4"
      >
        22.5%
      </dd>
    </div>
    <div>
      <dt
        class="c3"
      >
        total trades
        :
      </dt>
      <dd
        class="c4"
      >
        50
      </dd>
    </div>
    <div>
      <dt
        class="c3"
      >
        volume (USD)
        :
      </dt>
      <dd
        class="c4"
      >
        <span>
          $540,600.00
        </span>
      </dd>
    </div>
  </dl>
</div>
`;
